resources/labview_diff.py

```diff
@@ -56,5 +55,96 @@ def labview_path_from_year(year):
     return r"{0}\National Instruments\LabVIEW {1}\LabVIEW.exe".format(os.environ["ProgramFiles(x86)"], year)
 
 
+def export_repo(target_ref):
+    """
+    Export a copy of the repository at a given ref to a temporary directory.
+
+    :param target_ref: The ref you want to export, e.g. `origin\master`
+    :return: A temporaryfile.TemporaryDirectory containing the repository at the given ref
+    """
+
+    directory = tempfile.TemporaryDirectory()
+    shutil.copytree(".git", path.join(directory.name, ".git"))
+    subprocess.check_call(["git", "checkout", "-f", target_ref], cwd=directory.name)
+
+    @contextmanager
+    def cleanup_make_all_writeable(directory):
+        try:
+            yield directory
+        finally:
+            # tempfile.TemporaryDirectory has a bug where it fails when readonly files
+            # are present. Clearing the readonly flag manually fixes this.
+            # When https://bugs.python.org/issue26660 is fixed, this code can be removed,
+            # and the temporary directory can be returned directly
+            for root, dirs, files in os.walk(directory.name):
+                for file in files:
+                    os.chmod(root + "/" + file, stat.S_IWRITE)
+
+    return cleanup_make_all_writeable(directory)
+
+
+def get_changed_labview_files(target_ref):
+    """
+    Get LabVIEW files which have changed compared to the target ref.
+
+    :param target_ref: The git ref to check for changed files against
+    :return: Tuples of the form (status, filename) where status is either "A" or "M", depending on whether the file was added or modified.
+    """
+    diff_args = ["git", "diff", "--name-status", "--diff-filter=AM", target_ref, "HEAD"]
+    diff_output = subprocess.check_output(diff_args).decode("utf-8")
+
+    # https://regex101.com/r/EFVDVV/1
+    diff_regex = re.compile(r"^([AM])\s+(.*\.vi[tm]?)$", re.MULTILINE)
+
+    for match in re.finditer(diff_regex, diff_output):
+        yield match.group(1), match.group(2)
+
+
+def diff_repo(workspace, output_dir, target_branch, lv_version):
+    diffs = get_changed_labview_files(target_branch)
+
+    with export_repo(target_branch) as directory:
+        for status, filename in diffs:
+            if status == "A":
```

Comment:
"A" ?? this should probably be an enum (if status == difModes.Add.value)

Reply:
I was debating between adding a type and leaving it as a string. We're parsing this from git's output, and we get them back as "A" and "M". Adding a type would make it a little more robust. Good catch.